editor/src/uuiui/inputs/string-input.tsx

```diff
@@ -187,3 +144,54 @@ const LabelBelow = styled.label({
   textAlign: 'center',
   display: 'block',
 })
+
+export type HeadlessStringInputProps = React.InputHTMLAttributes<HTMLInputElement>
+
+export const HeadlessStringInput = React.forwardRef<HTMLInputElement, HeadlessStringInputProps>(
+  (props, propsRef) => {
+    const { disabled, onKeyDown, onFocus } = props
+
+    const ref = React.useRef<HTMLInputElement>(null)
+
+    const handleOnKeyDown = React.useCallback(
+      (e: React.KeyboardEvent<HTMLInputElement>) => {
+        if (onKeyDown != null) {
+          onKeyDown(e)
+        }
+        if (e.key === 'Escape' || e.key === 'Enter') {
+          e.preventDefault()
+          e.stopPropagation()
+          if (typeof ref !== 'function' && ref.current != null) {
```

Comment:
Can `ref` ever be a function? The type of `useRef` implies not.

Reply:
ahh, I just moved this code but that indeed looks wrong

Reply:
fixed!